### PR TITLE
refs #9572 - api changes for cli

### PR DESCRIPTION
**Pull request by orrabin**

### CODE REVIEW

File: app/controllers/api/v2/discovery_rules_controller.rb [hunk 1]
```diff
@@ -31,7 +31,7 @@ def show
           param :name, String, :required => true
           param :search, String, :required => true
           param :hostgroup_id, Integer, :required => true
-          param :hostname, String, :required => true
+          param :hostname, String
```
Comment (lzap) [inline app/controllers/api/v2/discovery_rules_controller.rb]: Why we do this? It was required, but blank was allowed.

File: app/controllers/api/v2/discovery_rules_controller.rb [hunk 2]
```diff
@@ -31,7 +31,7 @@ def show
...
           param :max_count, Integer
```
Comment (tstrachota) [inline app/controllers/api/v2/discovery_rules_controller.rb]: Isn't it confusing to have input parameter `max_count` and return the same value renamed as `hosts_limit`?

File: app/controllers/api/v2/discovery_rules_controller.rb [hunk 3]
```diff
@@ -52,7 +52,7 @@ def create
           param :name, String, :required => true
```
Comment (tstrachota) [inline app/controllers/api/v2/discovery_rules_controller.rb]: The whole `discovery_rule` hash could be replaced by `param_group :discovery_rule, :as => :update`.

**Comment by tstrachota** [comment]
I'm no expert in discovery, the api parameters might be obvious to others but it could be worth adding some descriptions e.g. to `search` and `priority`.

File: app/controllers/api/v2/discovery_rules_controller.rb (hunk 2 above)
Comment (orrabin) [inline app/controllers/api/v2/discovery_rules_controller.rb]: @tstrachota I didn't want to change the model which already uses `:max_count` but the UI shows `Hosts Limit` so I wanted the api to make sense

File: app/controllers/api/v2/discovery_rules_controller.rb (hunk 1 above)
Comment (orrabin) [inline app/controllers/api/v2/discovery_rules_controller.rb]: @lzap is it still required? I think we changed it once we allowed blank:
https://github.com/theforeman/foreman_discovery/blob/develop/app/models/discovery_rule.rb#L12

Comment (lzap) [inline app/controllers/api/v2/discovery_rules_controller.rb]: You are right, correct.

File: app/controllers/api/v2/discovery_rules_controller.rb [hunk 4]
```diff
@@ -29,9 +29,9 @@ def show
       def_param_group :discovery_rule do
         param :discovery_rule, Hash, :action_aware => true do
           param :name, String, :required => true
-          param :search, String, :required => true
-          param :hostgroup_id, Integer, :required => true
-          param :hostname, String, :required => true
+          param :search, String, :required => true, :desc => N_("query to match discovered hosts for the particular rule.")
```
Comment (orrabin) [inline app/controllers/api/v2/discovery_rules_controller.rb]: @lzap I tried to change a little the text you put in the documentation so it's shorter. Do you have a better explanation for search?

Comment (lzap) [inline app/controllers/api/v2/discovery_rules_controller.rb]: That is very fine.

File: app/controllers/api/v2/discovery_rules_controller.rb [hunk 5]
```diff
@@ -29,9 +29,9 @@ def show
...
+          param :hostgroup_id, Integer, :required => true, :desc => N_("puts the rules in order, low numbers go first. Must be greater then zero.")
+          param :hostname, String
           param :max_count, Integer
           param :priority, Integer
           param :enabled, :bool
```
Comment (lzap) [inline app/controllers/api/v2/discovery_rules_controller.rb]: Can you add descriptions for all of them? It's all documented. Hostname is also interesting, max_count as well...

**Comment by orrabin** [comment]
@lzap I added descriptions and rebased. Please check the descriptions are clear.

**Comment by lzap** [comment]
Nitpick - remove trailing dots where not necessary.

File: app/controllers/api/v2/discovery_rules_controller.rb [hunk 6]
```diff
@@ -28,13 +28,13 @@ def show
...
-          param :search, String, :required => true
-          param :hostgroup_id, Integer, :required => true
-          param :hostname, String, :required => true
-          param :max_count, Integer
-          param :priority, Integer
-          param :enabled, :bool
+          param :name, String, :required => true, :desc => N_("represents rule name shown to the users.")
+          param :search, String, :required => true, :desc => N_("query to match discovered hosts for the particular rule.")
+          param :hostgroup_id, Integer, :required => true, :desc => N_("the hostgroup that is used to auto provision a host")
+          param :hostname, String, :desc => N_("defines a pattern to assign human-readable hostnames to the matching hosts (default is macMACADDRESS)")
```
Comment (lzap) [inline app/controllers/api/v2/discovery_rules_controller.rb]: I'd remove the default explanation.

**Comment by lzap** [comment]
Ok otherwise.

**Comment by orrabin** [comment]
@lzap I removed the dots and default explanation

**Comment by lzap** [comment]
Thanks, merged as 4e6b799

**Pull request closed without merging**